### ISSUE TITLE
Require jsdoc for TypeScript class fields

### ISSUE THREAD

**Issue by saulotoledo** [gajus/eslint-plugin-jsdoc#558]
I suggest to add an option to the rule [`require-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc/#eslint-plugin-jsdoc-rules-require-jsdoc) to require documentation in TypeScript class fields.

## Motivation

The library has a good support to TypeScript. We are able to document almost everything, but class fields are not supported yet.

In TypeScript:

```typescript
class Animal {

  /**
   * This is an ECMAScript private field (see
   * https://www.typescriptlang.org/docs/handbook/classes.html#ecmascript-private-fields)
   */
  #name: string;

  /**
   * A private field in TypeScript.
   */
  private species: string;

  /**
   * A protected field in TypeScript.
   */
  protected subSpecies: string;

  /**
   * A public field in TypeScript.
   */
  public color: string;

  /**
   * Visibility is optional and we may also have annotations.
   */
  @SomeAnnotation('optionalParameter')
  tail: boolean;
}
```

## Current behavior

`require-jsdoc` cannot handle the example.

## Desired behavior

We could have an option (`FieldDefinition`?) such that requires the documentation above if true.

**Comment by saulotoledo** [comment]
@brettz9 This is somewhat related to the bug you fixed.

**Comment by brettz9** [comment]
This is actually already possible using `contexts` (with the definition name `ClassProperty`). (For future reference, you can discover the AST names for language features of interest using `@typescript-eslint/parser` as the parser with https://astexplorer.net/ ; these can be used within `contexts`, as can full esquery expressions using such definition names. See also https://github.com/gajus/eslint-plugin-jsdoc/#eslint-plugin-jsdoc-advanced-ast-and-selectors )

However, I went ahead and added your example as a test case in 3c03616304688f111e78e1bebe199175ca731e58 to demonstrate how it can be done.

Closing as that should address, but feel free to comment further. Thanks for the report!

**Comment by saulotoledo** [comment]
Thanks again @brettz9. I confirm that works for me and the configuration is very useful!
